Sources/PowerDiagram/PowerDiagram.swift

```diff
@@ -0,0 +1,72 @@
+import class Foundation.FileHandle
+import StringWidth
+import struct Foundation.Selector
+
+public struct PowerDiagram {
+  public struct Label: Comparable {
+    public let value: String
+    public let column: Int
+
+    public init(value: String, column: Int) {
+      self.value = value
+      self.column = column
+    }
+
+    static public func == (lhs: Self, rhs: Self) -> Bool {
+      lhs.column == rhs.column
+    }
+
+    static public func < (lhs: Self, rhs: Self) -> Bool {
+      lhs.column < rhs.column
+    }
+  }
+
+  private let mainLine: String
+  private let labels: [Label]
+
+  public init(mainLine: String, labels: [Label] ) {
+    self.mainLine = mainLine
+    self.labels = labels
+  }
+
+  public func render() -> String {
+    var labels = self.labels.sorted()
+
+    var message = "\(mainLine)\n"
+    renderFirstLine(into: &message)
+
+    while !labels.isEmpty {
+      var current = 0
+      var index = 0
+      while index < labels.count {
+        if index == labels.count - 1
+            || ((labels[index].column + stringWidth(labels[index].value) < labels[index + 1].column)
+                && labels[index].value.unicodeScalars.allSatisfy(\.isASCII))
+        {
+          align(&message, current: &current, column: labels[index].column, string: labels[index].value)
+          labels.remove(at: index)
+        } else {
+          align(&message, current: &current, column: labels[index].column, string: "│")
+          index += 1
+        }
+      }
+      message += "\n"
+    }
+
+    return message
+  }
+
+  private func renderFirstLine(into message: inout String) {
+    var current = 0
```

Comment:
Extracting this out prevents this `current` local being shadowed by the new `current` local var in the `while` loop above, which improves the clarify a bit.